### PR TITLE
feat(storage): Add dynamoDB Cloudformation templates

### PR DESCRIPTION
**Pull request by bvtujo**
This PR adds the cloudformation template necessary to create DDB tables on behalf of customers in storage init. 

Addresses #769 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bvtujo** [comment]
An example implementation is on the Go Playground [here](https://play.golang.org/p/aSgmSFe-xan).

The required data structures to execute this template look like this: 
```golang
type Attribute struct {
	Name     string
	DataType string
}

type LocalSecondary struct {
	Name         string
	PartitionKey string
	SortKey      string
}
type data struct {
	TableResourceName string
	TableName         string
	Attributes        []Attribute	
	PartitionKey      string
	SortKey           string
	HasLSI            bool
	LSIs              []LocalSecondary
}
```

File: templates/addons/ddb/cf.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{{.TableResourceName}}:
+  Type: AWS::DynamoDB::Table
+  Properties:
+    TableName: !Sub ${App}-${Env}-${Name}-{{.TableName}}
+    AttributeDefinitions:{{range .Attributes}}
+      - AttributeName: {{.Name}}
+        AttributeType: {{.DataType}}{{end}}
+    ProvisionedThroughput: 
+      ReadCapacityUnits: 2
+      WriteCapacityUnits: 2
```
Comment (kohidave) [inline templates/addons/ddb/cf.yml:10]: Maybe we can add `PAY_PER_REQUEST` as the billing mode and get rid of the provisioned throughput.

File: templates/addons/ddb/cf.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    KeySchema: 
+      - AttributeName: {{.PartitionKey}}
+        KeyType: HASH
+      - AttributeName: {{.SortKey}}
```
Comment (kohidave) [inline templates/addons/ddb/cf.yml:14]: I think we should add an if here, and only do the range key if someone provides it.

File: templates/addons/ddb/policy.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          Effect: Allow
+          Action:
+          - dynamodb:BatchGet*
+          - dynamodb:DescribeStream
+          - dynamodb:DescribeTable
+          - dynamodb:Get*
+          - dynamodb:Query
+          - dynamodb:Scan
+          - dynamodb:BatchWrite*
+          - dynamodb:CreateTable
```
Comment (kohidave) [inline templates/addons/ddb/policy.yml:20]: Can we do `Create*` so that folks can `CreateBackups` too?

File: templates/addons/ddb/outputs.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{{.TableResourceName}}TableName:
+  Description: "The name of the DynamoDB table for this application."
```
Comment (kohidave) [inline templates/addons/ddb/outputs.yml:2]:
```suggestion
  Description: "The name of this DynamoDB."
```

**Pull request merged**